src/FunderMaps.Core/UseCases/GeocoderUseCase.cs

```diff
@@ -42,5 +44,20 @@ public virtual async ValueTask<Address> GetAsync(string id)
         /// <param name="navigation">Recordset nagivation.</param>
         public virtual IAsyncEnumerable<Address> GetAllBySuggestionAsync(string query, INavigation navigation)
             => _addressRepository.GetBySearchQueryAsync(query, navigation);
+
+        /// <summary>
+        ///     Retrieve all addresses matching search query.
+        /// </summary>
+        /// <param name="query">Search query.</param>
+        /// <param name="navigation">Recordset nagivation.</param>
+        public async virtual IAsyncEnumerable<Address> GetAllBySuggestionWithBuildingAsync(string query, INavigation navigation)
+        {
+            await foreach (var address in GetAllBySuggestionAsync(query, navigation))
+            {
+                // FUTURE: This is a massive performance hit.
+                address.BuildingNavigation = await _buildingRepository.GetByIdAsync(address.BuildingId);
```

Comment:
Do we really need this? Isn't extending `Address` with a `building_geometry` field or something similar a more effective method of achieving this result? We can then return this from a database view.